packages/plugins/users-permissions/admin/src/pages/Roles/ListPage/index.js

```diff
@@ -1,195 +1,59 @@
-    },
-    [canDelete]
-  );
-
-  const getLinks = role => {
-    const links = [];
-
-    if (canUpdate) {
-      links.push({
-        icon: <FontAwesomeIcon icon="pencil-alt" />,
-        onClick: () => handleGoTo(role.id),
-      });
-    }
-    if (checkCanDeleteRole(role)) {
-      links.push({
-        icon: <FontAwesomeIcon icon="trash-alt" />,
-        onClick: e => {
-          e.preventDefault();
-          setModalDelete(role.id);
-          e.stopPropagation();
-        },
-      });
-    }
-
-    return links;
-  };
+  const pageTitle = formatMessage({
+    id: getTrad('HeaderNav.link.roles'),
+    defaultMessage: 'Roles',
+  });
 
   return (
-    <>
-      <Helmet title={formatMessage({ id: getTrad('page.title') })} />
-
-      <Header
-        icon
-        title={{
-          label: formatMessage({
-            id: 'Settings.roles.title',
-            defaultMessage: 'Roles & Permissions',
-          }),
-        }}
-        content={formatMessage({
-          id: 'Settings.roles.list.description',
-          defaultMessage: 'Define the roles and permissions for your users.',
+    <Layout>
+      <SettingsPageTitle name={pageTitle} />
+      <Main
+        labelledBy={formatMessage({
+          id: getTrad('HeaderNav.link.roles'),
+          defaultMessage: 'Roles',
         })}
-        actions={headerActions}
-        // Show a loader in the header while requesting data
-        isLoading={isLoading || isLoadingForPermissions}
-      />
-
-      <BaselineAlignment />
-      {canRead && (
-        <RoleListWrapper>
-          <List
-            title={formatMessage(
-              {
-                id: `Settings.roles.list.title${roles.length > 1 ? '.plural' : '.singular'}`,
-              },
-              { number: roles.length }
-            )}
-            items={roles}
-            isLoading={isLoading || isLoadingForPermissions}
-            customRowComponent={role => (
-              <RoleRow onClick={() => handleGoTo(role.id)} links={getLinks(role)} role={role} />
-            )}
-          />
-          {!roles && !isLoading && !isLoadingForPermissions && <EmptyRole />}
-          <PopUpWarning
-            isOpen={Boolean(modalToDelete)}
-            onConfirm={handleDelete}
-            onClosed={handleClosedModalDelete}
-            toggleModal={() => setModalDelete(null)}
-            isConfirmButtonLoading={showModalConfirmButtonLoading}
-          />
-        </RoleListWrapper>
-      )}
-    </>
+      >
+        <HeaderLayout
+          as="h1"
+          id="roles"
+          title={formatMessage({
+            id: 'Settings.roles.title',
+            defaultMessage: 'Roles',
+          })}
+          subtitle={formatMessage({
+            id: 'Settings.roles.list.description',
+            defaultMessage: 'List of roles',
+          })}
+          primaryAction={
+            <Button onClick={handleNewRoleClick} startIcon={<AddIcon />}>
```

Comment:
There is a permission that you need to apply here you can use the following component:


```
import { CheckPermissions } from '@strapi/helper-plugin';

primaryAction={<CheckPermissions permissions={permissions.createRole}><Button {...} /></CheckPermissions>}

```